src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -165,7 +175,7 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
 
                         //
                         // IContextualFeatureFilter
-                        if (useAppContext)
+                        if (useAppContext && ContextualFeatureFilterEvaluator.IsContextualFilter(filter, typeof(TContext)))
```

Comment:
Why is the `IsContextualFilter` function written like it is? Couldn't it just be `interfaceType.IsAssignableFrom(classType)` ? I know it's not a part of this PR, but curious.

Reply:
There is a staic method IsContextualFilter() of ContextualFeatureFilterEvaluator. 
In my previous commit, my code looked like: 
`if (useAppContext && ! (filter is IFeatureFilter))`
I think it is better to use IsContextualFilter() because it is more readable.